### DISCUSSION TITLE
Even more flexible edit link

### DISCUSSION THREAD

**Discussion by ailin-nemui** [pradyunsg/furo#471] [Ideas] [answered]
Currently, there is a huge switch in https://github.com/pradyunsg/sphinx-basic-ng/blob/main/src/sphinx_basic_ng/theme/basic-ng/components/edit-this-page.html to implement some edit links (https://github.com/pradyunsg/furo/discussions/114)

I would like to suggest that this be made more flexible, and I believe it would not come as a huge maintenance cost

Some sample code how it could be done, by adding a template setting variable:

```diff
diff --git a/src/furo/theme/furo/theme.conf b/src/furo/theme/furo/theme.conf
index c9f2eefd..1d39cc2e 100644
--- a/src/furo/theme/furo/theme.conf
+++ b/src/furo/theme/furo/theme.conf
@@ -23,6 +23,7 @@ sidebar_hide_name =
 footer_icons =
 top_of_page_button = edit
 # For components/edit-this-page.html
+source_repository_link =
 source_repository =
 source_branch =
 source_directory =
```

Then, in `src/furo/theme/furo/components/edit-this-page.html`:

```jinja2
  ...
{%- if theme_source_repository_link and page_source_suffix and not meta['generated'] -%}
  {% set url = theme_source_repository_link|format(pagename + page_source_suffix) %}
  {{ furo_edit_button(url) }}
{%- elif theme_source_repository -%}
  ...
```

Usage would also be rather simple: in `conf.py`

```py
html_theme_options['source_repository_link'] = 'https://my.repo.forge/myorg/myrepo/blob/mybranch/%s?plain=1'
```

**Comment by pradyunsg** [comment, marked as answer]
This will come with https://github.com/pradyunsg/sphinx-basic-ng/pull/34.

**Comment by marxin** [comment]
I would close this as resolved by #510.
